src/analyses/wrapperFunctionAnalysis.ml

```diff
@@ -131,8 +131,19 @@ module MallocWrapper : MCPSpec = struct
       CilType.Location.show loc
 
     let name_varinfo (t, node, c) =
-      Format.asprintf "(alloc@sid:%s@tid:%s(#%s))" (Node.show_id node) (ThreadLifted.show t) (UniqueCount.show c)
-
+        let uniq_count =
+          if UniqueCount.is_top c then
+            Format.dprintf ""
+          else
+            Format.dprintf "(#%s)" (UniqueCount.show c)
+        in
+        let tid =
+          if ThreadLifted.is_top t then
+            Format.dprintf ""
+          else
+            Format.dprintf "@tid:%s%t" (ThreadLifted.show t) uniq_count
+        in
+        Format.asprintf "(alloc@sid:%s%t)" (Node.show_id node) tid
```

Comment:
The statement IDs (`sid`) were added into alloc variable names for incremental analysis at some point. Are they actually required, @jerhard / @stilscher?

The one remaining problem in this PR is that cram tests fail on OSX because `sid`s are different. I would just remove them from the variable names altogether (races on those variables are reported for the right node locations anyway), but maybe some incremental logic requires them to match variables by name?

Without it, alloc variable names in the same thread would all have the same name (but different `vid` and location). For the user looking at Goblint race warnings, these numbers are meaningless clutter in the output. Even for us, they are only meaningful when looking at the HTML at the same time.

Reply:
Julian and me think that it should be possible to remove the `sid`  from the variable names. It was added when removing the location from the name, because locations can change in different program versions even without actual semantic changes. Since the malloc variable names are not part of the `NodeVarinfoMap` we can at least not think of a possible problem when removing the `sid` from the name.

Reply:
> Without it, alloc variable names in the same thread would all have the same name (but different vid and location). 

Do we print `vid`s by default everywhere we print names? Otherwise removing the `sid` is confusing because it will be harder to track the evolution of a variable across program states if both copies have the same name (I'm not sure if we can always rely on the order in such casses).

Reply:
This then is an argument that we should keep `sid`s regardless.

Reply:
The issue seems to be caused by the function `create_var` in `cilfacade.ml` because vars with the same name get the same hash and thus also the same `vid`. @sim642 has already questioned this function in https://github.com/goblint/analyzer/pull/1092#discussion_r1447083592.

Reply:
The string that is shown for a malloced-block does not necessarily have to be the same `vname` as the name; one could adapt the printing in `BaseType.Variables.show`. 

For HTML-output I would also be in favor of having at least one of the line or the sid visible.